src/catalog_api.cpp

```diff
@@ -75,30 +72,130 @@ static IRCAPIColumnDefinition ParseColumnDefinition(yyjson_val *column_def) {
+	}
+	endpoint_it->second = endpoint;
+}
+
 IRCAPITableCredentials IRCAPI::GetTableCredentials(ClientContext &context, IRCatalog &catalog, const string &schema, const string &table, IRCCredentials credentials) {
 	IRCAPITableCredentials result;
 	string api_result = GetTableMetadataCached(context, catalog, schema, table, catalog.secret_name);
 	std::unique_ptr<yyjson_doc, YyjsonDocDeleter> doc(ICUtils::api_result_to_doc(api_result));
 	auto *root = yyjson_doc_get_root(doc.get());
-	auto *warehouse_credentials = yyjson_obj_get(root, "config");
-	auto credential_size = yyjson_obj_size(warehouse_credentials);
 	auto catalog_credentials = IRCatalog::GetSecret(context, catalog.secret_name);
-	if (warehouse_credentials && credential_size > 0) {
-		result.key_id = IcebergUtils::TryGetStrFromObject(warehouse_credentials, "s3.access-key-id", false);
-		result.secret = IcebergUtils::TryGetStrFromObject(warehouse_credentials, "s3.secret-access-key",  false);
-		result.session_token = IcebergUtils::TryGetStrFromObject(warehouse_credentials, "s3.session-token", false);
-		if (catalog_credentials) {
-       		auto kv_secret = dynamic_cast<const KeyValueSecret &>(*catalog_credentials->secret);
-			auto region = kv_secret.TryGetValue("region").ToString();
-			result.region = region;
+
+	// Mapping from config key to a duckdb secret option
+
+	case_insensitive_map_t<Value> config_options;
+	if (catalog_credentials) {
+		auto kv_secret = dynamic_cast<const KeyValueSecret &>(*catalog_credentials->secret);
+		auto region = kv_secret.TryGetValue("region").ToString();
+		config_options["region"] = region;
+	}
+
+	auto *warehouse_credentials = yyjson_obj_get(root, "config");
+	ParseConfigOptions(warehouse_credentials, config_options);
+
+	if (!config_options.empty()) {
+		CreateSecretInfo info(OnCreateConflict::REPLACE_ON_CONFLICT, SecretPersistType::TEMPORARY);
+		info.options = config_options;
+		info.name = "PLACEHOLDER";
+		info.type = "s3";
+		info.provider = "config";
+		info.storage_type = "memory";
+		result.storage_credentials.push_back(info);
+	}
+
+	auto *storage_credentials = yyjson_obj_get(root, "storage-credentials");
+	auto storage_credentials_size = yyjson_arr_size(storage_credentials);
+	if (storage_credentials && storage_credentials_size > 0) {
+		yyjson_val *storage_credential;
+		size_t index, max;
+		yyjson_arr_foreach(storage_credentials, index, max, storage_credential) {
+			auto *sc_prefix = yyjson_obj_get(storage_credential, "prefix");
+			if (!sc_prefix) {
+				throw InternalException("required property 'prefix' is missing from the StorageCredential schema");
+			}
+
+			CreateSecretInfo create_secret_info(OnCreateConflict::REPLACE_ON_CONFLICT, SecretPersistType::TEMPORARY);
+			auto prefix_string = yyjson_get_str(sc_prefix);
+			if (!prefix_string) {
+				throw InternalException("property 'prefix' of StorageCredential is NULL");
+			}
+			create_secret_info.scope.push_back(string(prefix_string));
+			create_secret_info.type = "s3";
```

Comment:
This might need a name, because that's probably used for the conflict resolution.
if we have multiple "unnamed" secrets, they replace eachother